src/context/ErrorContext.jsx

```diff
@@ -0,0 +1,12 @@
+import { createContext, useState } from "react";
+
+export const ErrorContext = createContext();
+
+export const ErrorProvider = ({ children }) => {
```

Comment:
I would avoid having an error context for the same reasons you'd want to have a separate loading state for each request - if one error is used to update the error context and is therefore displaying on the screen, and the next error updates the error context, this will affect the other error displayed on the screen and both will potentially display the same error - this error context can only be used in one place for one error and therefore doesn't serve the purpose of the context and should be a state 👍